src/compile/mark/valueref.ts

```diff
@@ -80,10 +80,15 @@ export function midPoint(channel: Channel, channelDef: ChannelDef, scaleName: st
   defaultRef: VgValueRef | 'zeroOrMin' | 'zeroOrMax'): VgValueRef {
   // TODO: datum support
 
+  // FIXME: walk through the logic here
+
   if (channelDef) {
     /* istanbul ignore else */
     if (isFieldDef(channelDef)) {
       if (hasDiscreteDomain(scale.type)) {
+        if (channelDef.bin) {
+          return fieldRef(channelDef, scaleName, {binSuffix: 'start'});
```

Comment:
FYI, line 98 checks for bin too, so definitely needs some revision.  

I wonder if we should add new scale.hasBinnedDomain as a third case since bin is a mixed between the two.